### ISSUE TITLE
Tool is showing 0 upload and 0 download with a 1800000.0 ms ping

### ISSUE THREAD

**Issue by timeassistant** [sivel/speedtest-cli#695]
Hi there, I've had your script installed on 2 of my servers since before V 1.0.7 (that's the last one I had installed) (one of them seems to have been installed way back in 2016!)

It used to work perfectly in the past but today I tried to speed test the 2 servers and I get the excerpts pasted at the bottom of this message (note I ran multiple tests but for readability I am just showing the 2 tests to Vox (these were the auto chosen) and then the 2 extra tests to different locations.

Both of my servers are cPanel/WHM one is Centos 6 other is Centos 7. They are stuck on Python 2.6 and 2.7 respectively due to that being tied into core OS, not sure if that's relevant.

I have also updated the script to latest version and even ran a pip install over this after I removed the speedtest file. Still nothing. 

would appreciate some insight please - I do not see any errors being thrown up either but there's definitely an issue. 

The server are also hosted in different datacenters, one is a dedicated machine the other a VPS

`# speedtest-cli
Retrieving speedtest.net configuration...
Testing from my host (87.###.###.###)...
Retrieving speedtest.net server list...
Selecting best server based on ping...
Hosted by Vox Telecom (London) [1.65 km]: 1800000.0 ms
Testing download speed................................................................................
Download: 0.00 Mbit/s
Testing upload speed................................................................................`

`# speedtest-cli
Retrieving speedtest.net configuration...
Testing from my host (185.###.###.###)...
Retrieving speedtest.net server list...
Selecting best server based on ping...
Hosted by Vox Telecom (London) [2.56 km]: 1800000.0 ms
Testing download speed................................................................................
Download: 0.00 Mbit/s
Testing upload speed................................................................................................
Upload: 0.00 Mbit/s
`

`# speedtest-cli --server 16353
Retrieving speedtest.net configuration...
Testing from my host (185.###.###.###)...
Retrieving speedtest.net server list...
Retrieving information for the selected server...
Hosted by ITPS Ltd (Newcastle Upon Tyne) [399.19 km]: 1800000.0 ms
Testing download speed................................................................................
Download: 0.00 Mbit/s
Testing upload speed................................................................................................
Upload: 0.00 Mbit/s
`

`# speedtest-cli --server 838
Retrieving speedtest.net configuration...
Testing from my host (87.###.###.###)...
Retrieving speedtest.net server list...
Retrieving information for the selected server...
Hosted by XILO (Maidenhead) [41.18 km]: 1800000.0 ms
Testing download speed................................................................................
Download: 0.00 Mbit/s
Testing upload speed................................................................................................
Upload: 0.00 Mbit/s
`

would appreciate some help with this, thanks

just an additional, my servers are fine. One of them can be seen here downloading a 10gig file at just under 1gbps....  

![image](https://user-images.githubusercontent.com/18701905/73289792-3c23e600-41f5-11ea-97d3-fd0d353f7748.png)

**Comment by avanshee** [comment]
I too have the same issue. I have a RPI with access point capabilities running on it. I wanted capture the upload and download bandwidth  for both interfaces eth0 and wlan0.
I ran 
`speedtest-cli --simple` 
and I get the result as 
`Ping: 1800000.0 ms
Download: 0.00 Mbit/s
Upload: 0.00 Mbit/s`

Please, if someone could tell me what is going wrong here.

**Comment by sivel** [comment]
Duplicate of https://github.com/sivel/speedtest-cli/issues/567

**Comment by timeassistant** [comment]
@sivel  This is not a duplicate, as I stated in my first post my servers' connectivity is fine and I tried different test servers further out. Furthermore, both of my servers are hosted at different data centres operated by different companies... 

Please re-open and investigate this issue?

**Comment by sivel** [comment]
If you are getting a ping of 1800000.0ms, that explicitly means that the device where you are running speedtest-cli from, could not communicate with the speedtest server.  It is the same issue.

**Comment by timeassistant** [comment]
> 
> 
> If you are getting a ping of 1800000.0ms, that explicitly means that the device where you are running speedtest-cli from, could not communicate with the speedtest server. It is the same issue.

when you say could not communicate with speedtest server, do you mean the main server or the test server selected?

**Comment by timeassistant** [comment]
I can also ping speedtest.net from the servers too... 

[root@server ~]# ping speedtest.net
PING speedtest.net (151.101.194.219) 56(84) bytes of data.
64 bytes from 151.101.194.219: icmp_seq=1 ttl=57 time=0.515 ms
64 bytes from 151.101.194.219: icmp_seq=2 ttl=57 time=0.476 ms

I'm sorry but this does not seem like an issue being caused from either one of my servers @sivel 

If you still believe this is an issue with my 2 servers please provide a way to prove this?

**Comment by sivel** [comment]
> do you mean the main server or the test server selected?

The selected test server.

**Comment by sivel** [comment]
And more specifically, the http/https communication over the port that comes from `speedtest-servers.php`.  Which may not necessarily be 80/443.

**Comment by timeassistant** [comment]
@sivel  Ok thought that was what you meant.

I can ping and reach these test servers without using the speedtest-cli tool.  this is the Norway server at the top of the servers list

[root@server ~]# ping speedtest.oppinord.no
PING speedtest.oppinord.no (185.97.196.25) 56(84) bytes of data.
64 bytes from Speedtest.oppinord.no (185.97.196.25): icmp_seq=1 ttl=45 time=54.8 ms
64 bytes from Speedtest.oppinord.no (185.97.196.25): icmp_seq=2 ttl=45 time=54.9 ms
64 bytes from Speedtest.oppinord.no (185.97.196.25): icmp_seq=3 ttl=45 time=54.8 ms


<server url="http://speedtest.oppinord.no:8080/speedtest/upload.php" lat="70.6632" lon="23.6817" name="Hammerfest" country="Norway" cc="NO" sponsor="Hammerfest Energi Bredbånd AS" id="21239" host="speedtest.oppinord.no:8080"/>


Now I appreciate I can't explicitly ping port 8080  but again my servers CAN reach these servers. 
I ran an nmap test...


[root@genesis scripts]# nmap -p 8080 speedtest.oppinord.no

Starting Nmap 6.40 ( http://nmap.org ) at 2020-01-29 16:16 GMT
sendto in send_ip_packet_sd: sendto(4, packet, 40, 0, 185.97.196.25, 16) => Operation not permitted
Offending packet: TCP 87.117.247.59:61254 > 185.97.196.25:80 A ttl=48 id=4917 iplen=40  seq=0 win=1024
sendto in send_ip_packet_sd: sendto(4, packet, 44, 0, 185.97.196.25, 16) => Operation not permitted
Offending packet: TCP 87.117.247.59:61510 > 185.97.196.25:8080 S ttl=57 id=56503 iplen=44  seq=303620710 win=1024 <mss 1460>
sendto in send_ip_packet_sd: sendto(4, packet, 44, 0, 185.97.196.25, 16) => Operation not permitted
Offending packet: TCP 87.117.247.59:61511 > 185.97.196.25:8080 S ttl=53 id=2296 iplen=44  seq=303686247 win=1024 <mss 1460>
Nmap scan report for speedtest.oppinord.no (185.97.196.25)
Host is up (0.060s latency).
rDNS record for 185.97.196.25: Speedtest.oppinord.no
PORT     STATE    SERVICE
8080/tcp filtered http-proxy

Nmap done: 1 IP address (1 host up) scanned in 1.74 seconds



I mean can we look at other possibilities now? I feel I have provided sufficient evidence proving this is not an issue related or caused by either of my servers

thanks
